cli/flox/src/commands/auth.rs

```diff
@@ -74,25 +76,101 @@ pub async fn authorize(client: BasicClient) -> Result<Credential> {
 
     debug!("Device code details: {details:#?}");
 
-    eprintln!(
-        "Please visit {} in your browser",
-        details.verification_uri().as_str()
+    let url = details.user_code().secret().clone();
+    let prefilled_url = details
+        .verification_uri_complete()
+        .map(|uri| uri.secret())
+        .cloned();
+
+    eprintdoc! {"
+        First copy your one-time code: {code}
+
+        Press [enter] to open {url} in your browser.
+        ",
+        url = details.verification_uri().as_str(),
+        code = details.user_code().secret()
+    };
+
+    debug!(
+        "Waiting for user to enter code (timeout: {}s)",
+        details.expires_in().as_secs()
     );
-    eprintln!("And enter code: {}", details.user_code().secret());
 
-    let token_result = client
+    // in he background listen for `[enter]` key presses
+    // if the user presses enter, open the browser using the system default opener
+    // on linux this should be `xdg-open`
+    // on macos this should be `open`
+    //
+    // I'm not sure this is the best way to do this, but it works for now.
+    // Particularly, if the user opens the URL manually,
+    // there will be one less newline in the terminal than expected.
+    let done = Arc::new(AtomicBool::default());
+    let done_clone = done.clone();
+    tokio::task::spawn(async move {
+        loop {
+            if done_clone.load(Ordering::Relaxed) {
+                return;
+            }
+
+            match crossterm::event::poll(time::Duration::from_millis(100)) {
+                Err(_) => {
+                    info!("Could not read input. Please open the following URL manually: {url}",);
+                    return;
+                },
+                Ok(false) => {
+                    continue;
+                },
+                Ok(true) => {
+                    match crossterm::event::read() {
+                        Err(_) => {
+                            info!("Could not read input. Please open the following URL manually: {url}",);
+                            return;
+                        },
+                        Ok(crossterm::event::Event::Key(key))
+                            if key.code == crossterm::event::KeyCode::Enter =>
+                        {
+                            break;
+                        },
+                        _ => {
+                            continue;
+                        },
+                    }
+                },
+            }
+        }
+
+        let opener = match std::env::consts::OS {
```

Comment:
Blocker: It would be better to detect the presence of the opener before displaying the prompt so that you don't have to ask them to press enter to open, find out the opener doesn't exist, and then display an error message.